File: Luna.Models.Tasks.View/Card/CardView.cs
using Luna.Models.Tasks.Domain.Card;
using Luna.Models.Tasks.View.CardAttributes;
using Luna.Models.Users.View.Users;

namespace Luna.Models.Tasks.View.Card;

public class CardView
{
	public Guid Id { get; set; }

	public String Header { get; set; }

	public String? Content { get; set; }

	public String? Description { get; set; }

	public TypeView CardType { get; set; }

	public Guid CreatedUserId { get; set; }

	public DateTime CreatedTimestamp { get; set; }

	public DateTime? Deadline { get; set; }

	public CardView? PreviousCard { get; set; }

	public IEnumerable<CommentView> Comments { get; set; }

	public IEnumerable<TagView> CardTags { get; set; }

	//todo: add user model
	public IEnumerable<UserView> Users { get; set; }

	public StatusView Status { get; set; }

	public CardView(CardDomain cardDomain)
	{
		Id = cardDomain.Id;
		Header = cardDomain.Header;
		Content = cardDomain.Content;
		Description = cardDomain.Description;
		CardType = new TypeView(cardDomain.CardType);
		CreatedUserId = cardDomain.CreatedUserId;
		CreatedTimestamp = cardDomain.CreatedTimestamp;
		Deadline = cardDomain.Deadline;
		PreviousCard = cardDomain.PreviousCard == null ? null : new CardView(cardDomain.PreviousCard);
		Comments = cardDomain.Comments.Select(commentDomain => new CommentView(commentDomain));
		CardTags = cardDomain.CardTags.Select(tagDomain => new TagView(tagDomain.Tag));
		Users = cardDomain.Users.Select(c => new UserView(c.User)).ToList();
		Status = new StatusView(cardDomain.Status.Status);
	}

	public CardView
	(
		CardDomain cardDomain,
		TypeView cardType,
		IEnumerable<CommentView> comments,
		IEnumerable<TagView> tags,
		IEnumerable<UserView> users,
		StatusView status
	)
	{
		Id = cardDomain.Id;
		Header = cardDomain.Header;
		Content = cardDomain.Content;
		Description = cardDomain.Description;
		CreatedUserId = cardDomain.CreatedUserId;
		CreatedTimestamp = cardDomain.CreatedTimestamp;
		Deadline = cardDomain.Deadline;

		CardType = cardType;
		Comments = comments;
		CardTags = tags;
		Users = users;
		Status = status;
	}
}

File: luna/src/models/card/cardView.ts
import {Guid} from "guid-typescript";
import TypeView from "./typeView";
import CommentView from "./commentView";
import TagView from "./tagView";
import StatusView from "./statusView";

export default class CardView {

    constructor(
        id: Guid,
        header: string,
        content: string | null,
        description: string | null,
        cardType: TypeView,
        createdUserId: Guid,
        createdTimestamp: string,
        deadline: string | null,
        previousCard: CardView | null,
        comments: CommentView[],
        cardTags: TagView[],
        users: Guid[],
        statuses: StatusView[]
    ) {
        this.id = id;
        this.header = header;
        this.content = content;
        this.description = description;
        this.cardType = cardType;
        this.createdUserId = createdUserId;
        this.createdTimestamp = createdTimestamp;
        this.deadline = deadline;
        this.previousCard = previousCard;
        this.comments = comments;
        this.cardTags = cardTags;
        this.users = users;
        this.statuses = statuses;
    }

    public id: Guid;
    public header: string;
    public content: string | null;
    public description: string | null;
    public cardType: TypeView;
    public createdUserId: Guid;
    public createdTimestamp: string;
    public deadline: string | null;
    public previousCard: CardView | null;
    public comments: CommentView[];
    public cardTags: TagView[];
    public users: Guid[];
    public statuses: StatusView[];
}